tests/kernel_bundle/use_kernel_bundle_call_set_spec_const_before_use_kb_no_second_queue.cpp

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  This test calls sycl::handler::use_kernel_bundle after calling
+//  sycl::handler::set_specialization_constant without secondary queue.
+//
+//  The test verifies that the exception with sycl::errc::invalid was thrown.
+//
+*******************************************************************************/
+
+#include "../common/common.h"
+#include "use_kernel_bundle.h"
+
+#define TEST_NAME \
+  use_kernel_bundle_call_set_spec_const_before_use_kb_no_second_queue
+
+namespace TEST_NAMESPACE {
+using namespace sycl_cts;
+using namespace sycl_cts::tests::use_kernel_bundle;
+
+constexpr sycl::specialization_id<int> spec_const;
+
+/** @brief Call sycl::handler::set_specialization_constant after
+ *         sycl::handler::use_kernel_bundle with secondary queue
+ *  @param log sycl_cts::util::logger class object
+ *  @param ctx Context that will be used for constructing sycl::queue
+ *  @param kernel_bundle empty, non-empty or kernel_bundle with built-in kernels
+ */
+void run_verification(
+    sycl_cts::util::logger &log, const sycl::context &ctx,
+    sycl::kernel_bundle<sycl::bundle_state::executable> &kernel_bundle) {
+  sycl::queue queue(ctx, ctx.get_devices()[0]);
+  bool ex_was_thrown = false;
+  try {
+    queue.submit([&](sycl::handler &cgh) {
+      cgh.set_specialization_constant<spec_const>(0);
+      cgh.use_kernel_bundle(kernel_bundle);
+      cgh.single_task<kernel_for_kernel_bundle>([=]() {});
+    });
+  } catch (const sycl::exception &e) {
+    if (e.code() != sycl::errc::invalid) {
+      FAIL(log, unexpected_exception_msg);
+      throw;
+    }
+    ex_was_thrown = true;
```

Comment:
I guess the code could be simplified if we add a global `check_exception_with_invalid` to write
```c++
  check_exception_with_invalid([&] {
    queue.submit([&](sycl::handler &cgh) {
      cgh.set_specialization_constant<spec_const>(0);
      cgh.use_kernel_bundle(kernel_bundle);
      cgh.single_task<kernel_for_kernel_bundle>([=] {});
    });
  });
```
for the full test.

Reply:
@keryell It's a good idea, but in some cases we need call `set_specialization_constant` before calling `use_kernel_bundle`, in some cases we need do it after calling `use_kernel_bundle`.
In some cases we should use the secondary `queue`.

For all these cases we should have a few different `check_exception_with_invalid`. But in current solution we have `run_verification` function that  contains all things that will be done in this tests, but implementing a `check_exception_with_invalid` can add one encapsulation level that will hide some test actions. Therefore, I suggest do not change current test structure.

Reply:
OK. You know better your code than me. :-)